### ISSUE TITLE
Printing `fmp[z/q]_mat` not coherent

### ISSUE THREAD

**Issue by albinahlback** [wbhart/flint2#1056]
Printing `fmpz_mat` and `fmpq_mat` gives the following
```
**fmpz_mat_print_pretty**
[[0 0 0]
[0 0 0]
[0 0 0]
]

**fmpz_mat_print**
3 3  0 0 0 0 0 0 0 0 0

**fmpq_mat_print**
<3 x 3 matrix over Q>
[0, 0, 0]
[0, 0, 0]
[0, 0, 0]
```

Some questions:
- Should there be commas or not? I am fine with removing the commas in `fmpq_mat`.
- `fmpq_mat_print` is quite nicely printed. Should we call this `fmpq_mat_print_pretty` and remove `<n x n matrix over Q>`?
- Should an "ugly" version of printing for `fmpq_mat` be implemented, like the one for `fmpz_mat`?
- Should there be double square brackets or not, like seen in `fmpz_mat_print_pretty`?
- Should `fmpz_mat_print_pretty` print the last two brackets with a newline character in between?

Perhaps it goes for other types, I have not checked them out yet.

I suppose these are kind of breaking changes, but they need to be coherent at least (in my opinion).

**Comment by albinahlback** [comment]
Wrong documentation for `fmpq_poly_fprint_pretty` (also note `FILT` instead of `FILE`): https://flintlib.org/doc/fmpq_poly.html?highlight=print#c.fmpq_poly_fprint_pretty

**Comment by albinahlback** [comment]
I think the "ugly" versions are important and should be there to ease parsing. With that being said, I think `fmpz_[mat/vec]_print` does the correct thing, and thus think that `fmpq_mat_print` should print something very similar.

Thus, I think it remains to discuss the formatting of the output for the pretty printing. For matrix/vector-type objects, I am in favor of using one pair of brackets for each row (vectors can be interpretted as rows in order to not output so many newline-characters). I also think that commas are nice.

**Comment by wbhart** [comment]
I thought the issue was consistency. The only way to achieve that is to check what all the other functions are doing.

**Comment by albinahlback** [comment]
Yeah, we could either investigate how everything is printed and then do something about it. Or we could try to come up with a standardized format for everything based, and change it from there.
